### DISCUSSION TITLE
useActor with Fast Refresh in NextJS not working

### DISCUSSION THREAD

**Discussion by milanbgd011** [statelyai/xstate#2019] [Q&A]
For NextJS I found custom "useMachine" which works good with Fast Refresh in NextJS, it is located here:
https://github.com/davidkpiano/xstate/issues/995#issuecomment-636351806

When I used default useActor from @xstate/react lib, I get this error:
<img width="617" alt="Screenshot 2021-03-20 at 09 13 50" src="https://user-images.githubusercontent.com/20988958/111863690-6c7c9580-895d-11eb-8349-4396837537c7.png">

Fast Refresh is crucial for frontend stuff, especially for styling real-time components based on really complex machines, and useActor needs to work with Fast Refresh so that we can spawn many machines and have all working as intended.

I tried to look for custom useActor or modify existing custom useMachine, but failed.

Any pointers or information?

**Comment by Andarist** [comment]
Please always try to share a repro case in a runnable form - either by providing a git repository to clone or a codesandbox. OSS maintainers usually can't afford the time to set up the repro, even if exact steps are given.

**Reply by milanbgd011** [reply]
Ok, will make NextJS repo with all set, so it can be easily cloned and tested.

It seems that NextJS is breaking references among machines on refresh when saving irreversibly.

**Comment by milanbgd011** [comment]
I figured out a way to set all up properly, more in different Discussion here:
https://github.com/davidkpiano/xstate/discussions/2023
